packages/optivum-scrapper/src/timetable.ts

```diff
@@ -1,118 +1,237 @@
+            teacherShorts = new Map([...teacherShorts, ...parsedTable.teacherShorts]);
+            roomShorts = new Map([...roomShorts, ...parsedTable.roomShorts]);
+            commonGroups = new Map([...commonGroups, ...parsedTable.groups]);
+            parsedTable.interclassGroupIds.forEach((id) => {
+                const existingInterclassGroup = interclassGroups.get(id);
+                if (existingInterclassGroup) existingInterclassGroup.push(unit.id);
+                else interclassGroups.set(id, [unit.id]);
             });
-        } else {
-            const links = document.querySelectorAll('a');
-            units = [...links]
-                .map((link) => {
-                    const parsedLink = parseUnitLink(link);
-                    return parsedLink
-                        ? { id: parsedLink.id, type: parsedLink.type, fullName: link.textContent! }
-                        : null;
-                })
-                .filter(isDefined);
-        }
-        return units;
+
+            parsedTable.lessons.forEach((lesson) => Timetable.handleLesson(unit, lesson, lessons));
+        });
+
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (generatedDate === null) throw new Error('Missing generated date');
+
+        return formatResult(
+            days,
+            timeSlots,
+            classShorts,
+            teacherShorts,
+            roomShorts,
+            unitFullNames,
+            interclassGroups,
+            subjects,
+            commonGroups,
+            lessons,
+            generatedDate,
+            validFrom,
+        );
+    }
+
+    private static handleLesson(unit: Unit, lesson: LessonTimeSlot, lessons: DefaultsMap<string, ClientLesson[]>) {
+        const timeSlotLessons = lessons.get(`${lesson.dayIndex.toString()}|${lesson.timeSlotIndex.toString()}`);
+
+        if (lesson.type === 'comment') return timeSlotLessons.push({ type: 'comment', comment: lesson.comment });
+
+        const isMatchingLesson = (existingLesson: ClientLesson) => {
+            // Comment
+            if (existingLesson.type === 'comment') return;
+
+            // Lessons without interclass groups (by teacher)
+            if (lesson.teacherId !== null && existingLesson.teacherId !== null)
+                return lesson.teacherId === existingLesson.teacherId;
+
+            // Interclass groups
+            if (lesson.subjectId !== existingLesson.subjectId) return;
+            // Interclass groups: class tables
+            if (lesson.interclassGroupId !== null && existingLesson.interclassGroupId !== null)
+                return lesson.interclassGroupId === existingLesson.interclassGroupId;
+            // Interclass groups: teacher tables
+            if (unit.type !== UnitType.CLASS)
+                return [...existingLesson.classIds.values()].find((classId) => lesson.classIds.has(classId));
+        };
```

Comment:
Obecnie czasami ta funkcja zwraca `undefined`

```suggestion
        const isMatchingLesson = (existingLesson: ClientLesson): boolean => {
            // Comment
            if (existingLesson.type === 'comment') return false;

            // Lessons without interclass groups (by teacher)
            if (lesson.teacherId !== null && existingLesson.teacherId !== null)
                return lesson.teacherId === existingLesson.teacherId;

            // Interclass groups
            if (lesson.subjectId !== existingLesson.subjectId) return false;
            // Interclass groups: class tables
            if (lesson.interclassGroupId !== null && existingLesson.interclassGroupId !== null)
                return lesson.interclassGroupId === existingLesson.interclassGroupId;
            // Interclass groups: teacher tables
            if (unit.type !== UnitType.CLASS)
                return [...existingLesson.classIds.values()].find((classId) => lesson.classIds.has(classId));
            
            return false;
        };
```